storybookjs/storybook
# CLI: Add pnpm support

Issue: https://github.com/storybookjs/storybook/issues/19412

## What I did

This adds support for the use of pnpm during CLI operations.  I kept the same strategy as was used for npm, giving it a `--use-pnpm` flag to force its use.  Otherwise, we'll check if it's installed and look for a `pnpm-lock.yaml` file to determine if we should use it.

Note: this also adds support for forcing npm (as well as pnpm) for `sb automigrate`, which was missing before.

## How to test


I didn't add any end-to-end style test here, but it is possible to verify it's working manually:

```
# Compile the branch
yarn task --task compile --start-from compile

# from somewhere else outside of the storybook repo:
pnpm create vite@latest --template=react

# cd into your new project and run
pnpm install

# Initialize storybook
node ../../../storybook/storybook/code/lib/cli/dist/generate.js init
```

You should see pnpm being used to install dependencies.  Without the changes in this PR, the installation step would fail.
